### PR TITLE
Correctly measure duration of Query

### PR DESCRIPTION
**Pull request by arielshaqed**
Measure all the way through reading Rows.

---
This change is [Reviewable](https://reviewable.io/reviews/treeverse/lakefs/1042)

### CODE REVIEW

**Comment by arielshaqed** [comment]
Sample log line:
```log
INFO   [2020-12-10T17:18:36+02:00]db/traced_rows.go:34 db.(*LoggedRows).logDuration rows done                                     args="[foo ]" duration=2.234833ms query="\n\t    WITH resolved_policies_view AS (\n                SELECT auth_policies.id, auth_policies.created_at, auth_policies.display_name, auth_policies.statement, auth_users.display_name AS user_display_name\n                FROM auth_policies INNER JOIN\n                     auth_user_policies ON (auth_policies.id = auth_user_policies.policy_id) INNER JOIN\n\t\t     auth_users ON (auth_users.id = auth_user_policies.user_id)\n                UNION\n\t\tSELECT auth_policies.id, auth_policies.created_at, auth_policies.display_name, auth_policies.statement, auth_users.display_name AS user_display_name\n\t\tFROM auth_policies INNER JOIN\n\t\t     auth_group_policies ON (auth_policies.id = auth_group_policies.policy_id) INNER JOIN\n\t\t     auth_groups ON (auth_groups.id = auth_group_policies.group_id) INNER JOIN\n\t\t     auth_user_groups ON (auth_user_groups.group_id = auth_groups.id) INNER JOIN\n\t\t     auth_users ON (auth_users.id = auth_user_groups.user_id)\n\t    ) SELECT id, created_at, display_name, statement FROM resolved_policies_view WHERE user_display_name = $1 AND display_name > $2 ORDER BY display_name" type="start query"
```

File: db/traced_rows.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
+package db
```
Comment (nopcoder) [inline db/traced_rows.go:1]: align the filename with the type LoggedRows <-> traced_rows

File: db/traced_rows.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+// LoggedRows is a pgx.Rows that wraps and traces another pgx.Rows.
+type LoggedRows struct {
+	pgx.Rows
+	Start  time.Time
+	Closed bool
+	l      logging.Logger
+}
+
+var compileCheck pgx.Rows = &LoggedRows{}
```
Comment (nopcoder) [inline db/traced_rows.go:21]:
```suggestion
var _ pgx.Rows = &LoggedRows{}
```

**Review by nopcoder — CHANGES_REQUESTED** [review]
very cool

File: db/traced_rows.go (hunk 1 above)
Comment (arielshaqed) [reply db/traced_rows.go]: Done!

File: db/traced_rows.go (hunk 2 above)
Comment (arielshaqed) [reply db/traced_rows.go]: Done!

**Review by arielshaqed — COMMENTED** [review]
Thanks!

**Review by nopcoder — APPROVED** [review]
> Reviewed 1 of 4 files at r1.
> *[Reviewable](https://reviewable.io/reviews/treeverse/lakefs/1042)* status: 1 of 4 files reviewed, 2 unresolved discussions (waiting on @nopcoder)

**Pull request merged**